### PR TITLE
(PrincipalsDelete) Add delete button for groups

### PR DESCRIPTION
**Pull request by mrvisser**
Adds a delete button to leverage the back-end group delete work made available in https://github.com/oaeproject/Hilary/pull/1149

### CODE REVIEW

File: shared/oae/api/oae.api.group.js [hunk 1]
```diff
@@ -280,4 +280,32 @@ define(['exports', 'jquery', 'underscore', 'oae.api.util'], function(exports, $,
             }
         });
     };
+
+    /**
+     * Delete a group as the currently authenticated user.
```
Comment (nicolaasmatthijs) [inline shared/oae/api/oae.api.group.js]: We can just say `Delete a group` here. The join and leave function just have the `as the currently authenticated user` bit as it makes more sense there

**Comment by nicolaasmatthijs** [comment]
Left one small bit of feedback. Ready to merge after that. Back to @mrvisser

**Comment by nicolaasmatthijs** [comment]
We usually have a button in the library as well to allow deleting multiple items at once. However, I'm thinking that we might be better off not offering that for groups.

**Comment by mrvisser** [comment]
Feedback addressed, back to @nicolaasmatthijs . I suppose this should go `On Hold` until the back-end side of this is ready for merge as well.

**Comment by nicolaasmatthijs** [comment]
Ready for merging. Putting on hold until oaeproject/Hilary#1149 is ready to merge

**Comment by mrvisser** [comment]
I totally forgot about the library. But glad we don't need it :)

**Pull request merged**